testdata/p4_16_samples_outputs/nested_if_lvalue_dependencies-midend.p4

```diff
@@ -0,0 +1,99 @@
+#include <core.p4>
+#include <v1model.p4>
+
+typedef bit<48> macAddr_t;
+typedef bit<32> ip4Addr_t;
+header ethernet_t {
+    macAddr_t dstAddr;
+    macAddr_t srcAddr;
+    bit<16>   etherType;
+}
+
+header ipv4_t {
+    bit<4>    version;
+    bit<4>    ihl;
+    bit<8>    diffserv;
+    bit<16>   totalLen;
+    bit<16>   identification;
+    bit<3>    flags;
+    bit<13>   fragOffset;
+    bit<8>    ttl;
+    bit<8>    protocol;
+    bit<16>   hdrChecksum;
+    ip4Addr_t srcAddr;
+    ip4Addr_t dstAddr;
+}
+
+struct metadata {
+}
+
+struct headers {
+    ethernet_t ethernet;
+    ipv4_t     ipv4;
+}
+
+parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    state start {
+        packet.extract<ethernet_t>(hdr.ethernet);
+        transition select(hdr.ethernet.etherType) {
+            16w0x800: parse_ipv4;
+            default: accept;
+        }
+    }
+    state parse_ipv4 {
+        packet.extract<ipv4_t>(hdr.ipv4);
+        transition select(hdr.ipv4.protocol) {
+            default: accept;
+        }
+    }
+}
+
+control MyVerifyChecksum(inout headers hdr, inout metadata meta) {
+    apply {
+    }
+}
+
+control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_0() {
+    }
+    @name("MyIngress.ipv4_forward") action ipv4_forward() {
+        hdr.ipv4.totalLen = (hdr.ipv4.identification > 16w0 ? (hdr.ipv4.identification > 16w0 ? (hdr.ipv4.identification > 16w0 ? 16w1 : hdr.ipv4.identification) + (hdr.ipv4.identification > 16w0 ? 16w2 : hdr.ipv4.hdrChecksum) + 16w3 : hdr.ipv4.totalLen) + (hdr.ipv4.identification > 16w0 ? 16w5 : (hdr.ipv4.identification > 16w0 ? 16w1 : hdr.ipv4.identification)) + (hdr.ipv4.identification > 16w0 ? 16w4 : (hdr.ipv4.identification > 16w0 ? 16w2 : hdr.ipv4.hdrChecksum)) + 16w13 : (hdr.ipv4.identification > 16w0 ? (hdr.ipv4.identification > 16w0 ? 16w1 : hdr.ipv4.identification) + (hdr.ipv4.identification > 16w0 ? 16w2 : hdr.ipv4.hdrChecksum) + 16w3 : hdr.ipv4.totalLen)) + (hdr.ipv4.identification > 16w0 ? 16w5 : (hdr.ipv4.identification > 16w0 ? 16w1 : hdr.ipv4.identification)) + (hdr.ipv4.identification > 16w0 ? 16w4 : (hdr.ipv4.identification > 16w0 ? 16w2 : hdr.ipv4.hdrChecksum));
```

Comment:
Is this cost exponential in the number of consecutive if statements?

Reply:
For the consecutive if statements like:
`if(c0) x = 0;
if(c1) x=1; `
_Predication_ will make: 
`x = c0 ? 0 : x;
x = c1 ? 1 : x;`
Next midend pass *LocalCopyProp* will merge two assignments to:
`x = c1 ? 1 : ( c0 ? 0 : x );`
Now, this pass is optimized for the nested if-else statements, which will merge assignments before _LocalCopyProp_ pass.